src/platform/amd/renoir/lib/clk.c

```diff
@@ -0,0 +1,141 @@
+* acp_reg_read_via_smn()
+*/
+/**  \note      The \b acp_reg_read_via_smn function details are provided in
+*               acp_hal.h file.
+*
+*****************************************************************************/
+int acp_reg_read_via_smn (uint32_t	reg_offset,
+			 uint32_t	reg_value,
+			 uint32_t	size)
+{
+	uint32_t	smnClientBaseAddr    = (reg_offset >> 10);
+	uint32_t	regionStartAddr = (smnClientBaseAddr << 10);
+	acp_srbm_cycle_sts_t  acpSrbmCycleSts;
+	/*Configuring the MP1 Aperture Id in SRB_client_config register*/
+	io_reg_write((PU_REGISTER_BASE + ACP_SRBM_CLIENT_CONFIG), MP1_APERTURE_ID);
+	/*Configuring the base address of MP1*/
+	io_reg_write((PU_REGISTER_BASE + ACP_SRBM_CLIENT_BASE_ADDR), smnClientBaseAddr);
+	/*dummy read to the to the address offset 0x3400*/
+	/*(Master config offset to access register outside of ACP)*/
+	reg_value = (uint32_t)io_reg_read(PU_REGISTER_BASE + (ACP_MASTER_REG_ACCESS_ADDRESS + reg_offset - regionStartAddr + ACP_FIRST_REG_OFFSET));
+	/*reading status register to check above read request is completed or not*/
+	acpSrbmCycleSts = (acp_srbm_cycle_sts_t)io_reg_read(PU_REGISTER_BASE + ACP_SRBM_CYCLE_STS);
+	/*Waiting for status to be set to '0'*/
+	while (acpSrbmCycleSts.bits.srbm_clients_sts != 0)
+		acpSrbmCycleSts = (acp_srbm_cycle_sts_t)io_reg_read(PU_REGISTER_BASE + ACP_SRBM_CYCLE_STS);
+	/*when status is '0' from above read the data from data register*/
+	reg_value = (uint32_t) io_reg_read(PU_REGISTER_BASE + ACP_SRBM_Client_RDDATA);
+	return reg_value;
+}
+
+/*****************************************************************************\
+* acp_reg_write_via_smn()
+*/
+/**  \note      The \b acp_reg_write_via_smn function details are provided in
+*               acp_hal.h file.
+*
+*****************************************************************************/
+void acp_reg_write_via_smn(uint32_t	reg_offset,
+			   uint32_t	value,
+			   uint32_t	size)
+{
+	uint32_t	smnClientBaseAddr    = (reg_offset >> 10);
+	uint32_t	regionStartAddr = (smnClientBaseAddr << 10);
+	acp_srbm_cycle_sts_t      acpSrbmCycleSts;
+	/*Configuring the MP1 Aperture Id in SRB_client_config register*/
+	io_reg_write((PU_REGISTER_BASE + ACP_SRBM_CLIENT_CONFIG), MP1_APERTURE_ID);
+	/*Configuring the base address of MP1*/
+	io_reg_write((PU_REGISTER_BASE + ACP_SRBM_CLIENT_BASE_ADDR), smnClientBaseAddr);
+	/*write to the address offset 0x3400(Master config offset to access register outside of ACP)*/
+	io_reg_write((PU_REGISTER_BASE + (ACP_MASTER_REG_ACCESS_ADDRESS + reg_offset - regionStartAddr + ACP_FIRST_REG_OFFSET)), value);
+	/* reading status register to check above read request is completed or not*/
+	acpSrbmCycleSts = (acp_srbm_cycle_sts_t)io_reg_read(PU_REGISTER_BASE + ACP_SRBM_CYCLE_STS);
+	/*Waiting for status to be set to '0'*/
+	while (acpSrbmCycleSts.bits.srbm_clients_sts != 0)
+		acpSrbmCycleSts = (acp_srbm_cycle_sts_t)io_reg_read(PU_REGISTER_BASE + ACP_SRBM_CYCLE_STS);
```

Comment:
looping forever on HW state change. Needs a timeout and error print.

Reply:
Addressed the Review Comment. Done